spark/src/main/scala/org/apache/spark/sql/delta/UniversalFormat.scala

```diff
@@ -144,8 +145,35 @@ object UniversalFormat extends DeltaLogging {
     }
   }
 
-}
+  val ICEBERG_TABLE_TYPE_KEY = "table_type"
+
+  /**
+   * Update CatalogTable to mark it readable by other table readers (iceberg for now).
+   * This method ensures 'table_type' = 'ICEBERG' when uniform is enabled,
+   * and ensure table_type is not 'ICEBERG' when uniform is not enabled
+   * If the key has other values than 'ICEBERG', this method will not touch it for compatibility
+   *
+   * @param table    catalogTable before change
+   * @param metadata snapshot metadata
+   * @return the converted catalog, or None if no change is made
+   */
+  def enforceSupportInCatalog(table: CatalogTable, metadata: Metadata): Option[CatalogTable] = {
+    val icebergInCatalog = table.properties.get(ICEBERG_TABLE_TYPE_KEY) match {
+      case Some(value) => value.equalsIgnoreCase(ICEBERG_FORMAT)
+      case _ => false
+    }
 
+    (icebergEnabled(metadata), icebergInCatalog) match {
+      case (true, false) =>
+        Some(table.copy(properties = table.properties
+          + (ICEBERG_TABLE_TYPE_KEY -> ICEBERG_FORMAT)))
+      case (false, true) =>
+        Some(table.copy(properties =
+          table.properties - ICEBERG_TABLE_TYPE_KEY))
+      case _ => None
```

Comment:
(false, false) case seems safe enough to ignore, but what about (true, true)?
Does that just mean the table property is already present, so nothing more to do?